### ISSUE TITLE
Plugin crashes to segmentation violation when called without IPAM configuration

### ISSUE THREAD

**Issue by kangasta** [Billy99/user-space-net-plugin#7]
In my system, the current implementation is failing to segmentation violation. Stacktrace below. I'm using Go 1.10.3 and running the plugin from Kubernetes 1.11 and latest multus.

```txt
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4e1eb1]
goroutine 1 [running, locked to thread]:
github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/types/current.(*Result).GetAsVersion(0x0, 0xc42009e448, 0x5, 0x0, 0x0, 0x0, 0xc4200a61c0)
        /root/go/src/github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/types/current/types.go:200 +0x51
github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/types.PrintResult(0x5cb500, 0x0, 0xc42009e448, 0x5, 0x0, 0x0)
        /root/go/src/github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/types/types.go:100 +0x49
main.cmdAdd(0xc4200cc380, 0xc42009e288, 0x5)
        /root/go/src/github.com/Billy99/user-space-net-plugin/userspace/userspace.go:140 +0x21b
github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/skel.(*dispatcher).checkVersionAndCall(0xc4200a6180, 0xc4200cc380, 0x5ca380, 0xc42008c510, 0x5b5560, 0x0, 0x411fc8)
        /root/go/src/github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/skel/skel.go:162 +0x187
github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/skel.(*dispatcher).pluginMain(0xc4200a6180, 0x5b5560, 0x5b5568, 0x5ca380, 0xc42008c510, 0x593b00)
        /root/go/src/github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/skel/skel.go:173 +0x28a
github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/skel.PluginMainWithError(0x5b5560, 0x5b5568, 0x5ca380, 0xc42008c510, 0xc42005bf78)
        /root/go/src/github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/skel/skel.go:210 +0xe8
github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/skel.PluginMain(0x5b5560, 0x5b5568, 0x5ca380, 0xc42008c510)
        /root/go/src/github.com/Billy99/user-space-net-plugin/vendor/github.com/containernetworking/cni/pkg/skel/skel.go:222 +0x4d
main.main()
        /root/go/src/github.com/Billy99/user-space-net-plugin/userspace/userspace.go:230 +0x51
```

I was able to bypass this with:

```diff
diff --git a/userspace/userspace.go b/userspace/userspace.go
index ceebd7d..9745e14 100644
--- a/userspace/userspace.go
+++ b/userspace/userspace.go
@@ -20,7 +20,6 @@ import (
        "runtime"
 
        "github.com/containernetworking/cni/pkg/skel"
-       cnitypes "github.com/containernetworking/cni/pkg/types"
        "github.com/containernetworking/cni/pkg/types/current"
        cniSpecVersion "github.com/containernetworking/cni/pkg/version"
        "github.com/containernetworking/plugins/pkg/ip"
@@ -137,7 +136,7 @@ func cmdAdd(args *skel.CmdArgs) error {
                return err
        }
 
-       return cnitypes.PrintResult(result, netConf.CNIVersion)
+       return result.Print()
 }
 
 func cmdDel(args *skel.CmdArgs) error {
```

Not sure if this would be the correct way to go as the versioning is left out from the printing of the result. However, after this change and removing prints from OVS related packages (#3) I got a pod up and packets flowing though two vhostuser ports (with two NetworkAttachment CRDs and OVS as the engine).

**Comment by rkamudhan** [comment]
@kangasta We check build internally. I was working fine. I will check it again..

**Comment by kangasta** [comment]
Found the error now: I had omitted IPAM configuration from the NetworkAttachmentDefinition which, since the value for result is not initialized by the plugin itself but only in the IPAM plugin call, causes the result to be nil when `cnitypes.PrintResult(...)` is called. `result.Print` seems to work with nil input, whereas the `cnitypes.PrintResult(...)` crashes as it tries to actually use the nil pointer.

So I think the solution could be to add some kind of initialization for the result.
